### PR TITLE
Add overall help guide

### PR DESCRIPTION
**Pull request by richfitz**
Having looked for docs recently, this is something that I thought we could use

### CODE REVIEW

File: vignettes/guide.Rmd [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+`odin` has grown from just one package to a series of packages; this document aims to help orient you to the right place to understand how pieces fit together and where to find help.
+
+# The packages
+
+* [`odin`](https://mrc-ide.github.io/odin) contains the domain specific language (DSL), and the original implementation suitable for simple use.  It can compile continuous time (ODE) and discrete time models to C, R or JavaScript. It contains no ability to do inference.
+* [`dust`](https://mrc-ide.github.io/dust) contains an engine for running odin-style models in parallel. It supports both continuous time and discrete time models, but does not "know" anything about odin. This is the lower level support that many users will end up engaging with eventually.
```
Comment (weshinsley) [inline vignettes/guide.Rmd:19]: Just on the odin/dust relationship - 'dust doesn't "know" anything about odin' - but can run 'odin-style' models - but our standard use (I think) is to use dust to run odin models via odin.dust? Maybe I'm after a better understanding of what 'odin-style' might mean, if it's not pure odin itself.

**Review by weshinsley — APPROVED** [review]
Two tiny typos - and one comment where maybe I need to play with odin/dust separately...

File: vignettes/guide.Rmd [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* [`odin.dust`](https://mrc-ide.github.io/odin.dust) is a package for compiling `odin` models to work with `dust`. It will likely merge into `odin` once `dust` is on CRAN.
```
[the remark this answers is not in the document]
Comment (richfitz) [reply vignettes/guide.Rmd:20]: The stoppers for this atm at the autodiff and mpi work :)

File: vignettes/guide.Rmd (hunk 1 above)
Comment (richfitz) [reply vignettes/guide.Rmd:19]: This is talked about a little in the dust design vignette - dust provides a compilation target for odin models

**Review by r-ash — APPROVED** [review]
Nice! This looks very useful

**Pull request merged**